demo/demo.js

```diff
@@ -0,0 +1,337 @@
+    const selectedShippingType = document.querySelector(
+      '.shipping-type-option input[type=\'radio\']:checked');
+    if (selectedShippingType && selectedShippingType.value !== 'default') {
+      options.shippingType = selectedShippingType.value;
+    }
+
+    console.log(options);
+
+    // Why is this an array of an object with supportedMethods?
+    const supportedInstruments = [basicCards];
+    const details = {
+      displayItems: displayItemsFromUI,
+      // Excluding total will result in an error - it's a required field.
+      total: totalFromUI,
+      shippingOptions: shippingOptionsFromUI,
+    };
+
+    const paymentRequest = new PaymentRequest(
+      supportedInstruments,
+      details,
+      options);
+
+    paymentRequest.addEventListener(
+      'shippingaddresschange',
+      (event) => this.onShippingAddressChange(event, details)
+    );
+    paymentRequest.addEventListener(
+      'shippingoptionchange',
+      (event) => this.onShippingOptionChange(event, details)
+    );
+
+    paymentRequest.show()
+    .then((result) => {
+      // Process the payment
+      const data = {};
+      data.methodName = result.methodName;
+      data.details = result.details;
+
+      // $("#result").text(JSON.stringify(data, null, 2));
+      return result.complete('success');
+    })
+    .catch((err) => {
+      console.group(
+        'The promise from `paymentRequest.show()` rejected.');
+      console.warn('This is normally due to the user closing or cancelling ' +
+        'the payment request UI.');
+      console.warn(`The error message received was: '${err.message}'`);
+      console.error(err);
+      console.groupEnd();
+    });
+  }
+
+  onShippingAddressChange(event, previousDetails) {
+    const paymentRequest = event.target;
+    console.log(`Received a 'shippingaddresschange' event, change to: `,
+      paymentRequest.shippingAddress);
+
+    // The response needs to be details + at least one shipping option.
+    // If no shipping option is defined then the API treats it as the
+    // address is invalid.
+    // TODO: Can you define a custom error "We only ship to US addresses..."
+
+    // Assume every is valid in this demo.
+
+    event.updateWith(previousDetails);
+  }
+
+  // This event has nothing to do with Payment Request - just to explain
+  // the UI / behavior a little bit.
+  onEnableShippingChange() {
+    const isEnabled = document.querySelector('#checkbox-shipping').checked;
+    const shippingEnabledUI =
+      document.querySelector('.depends-on-shipping-enabled');
+    if (isEnabled) {
+      shippingEnabledUI.classList.remove('is-disabled');
+    } else {
+      shippingEnabledUI.classList.add('is-disabled');
+    }
+  }
+
+  onShippingOptionChange(event, previousDetails) {
+    const paymentRequest = event.target;
+    console.log(`Received a 'shippingoptionchange' event, change to: `,
+      paymentRequest.shippingOption);
+
+    previousDetails.shippingOptions.forEach((shippingOption) => {
```

Comment:
I'm torn.

The total is static, as are the display items. Doing anything for the shipping cost feels counter intuitive to this.

Personally I'd rather force people to add display items, calculate the total from that and then add shipping to the total as well.